primitives/src/host_functions/native_rng.rs

```diff
@@ -0,0 +1,130 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymathNetwork/Polymesh).
+// Copyright (c) 2020 Polymath
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use rand_core::{CryptoRng, Error, RngCore};
+use sp_runtime_interface::runtime_interface;
+
+#[cfg(not(feature = "std"))]
+use core::num::NonZeroU32;
+
+#[cfg(feature = "std")]
+use rand::rngs::OsRng;
+
+/// Host functions to allow native access to OS RNG from Wasm.
+///
+/// # Change Log
+/// - Runtime Spec version 1005. It is incompatible with previous versions, and it requires a
+/// binary update.
+///
+/// # TODO
+///  - Use lazy static RNG object to mitigate performance impact during the `OsRng` object
+///  creation. We should double-check the security implications of having a global RNG in a
+///  multi-threading environment.
+#[runtime_interface]
+pub trait NativeRng {
+    /// Returns the next random `u32` generated by `OsRng`.
+    fn next_u32() -> u32 {
+        OsRng::default().next_u32()
```

Comment:
Is this a permanent solution or can we use BABE's VRF to seed a PRNG?

Reply:
We have had to recovery this interface to query historic data on PME, where we've found the following error:
```
Error: -32603: Unknown error occurred: Client, VersionInvalid, Other, \"Instantiation: Export ext_native_rng_fill_bytes_version_1 not found\
```
A part of that, any other possible implementations would be nice. The first thought was to use the `randomness_collective_flip` pallet + the extrinsic index.